src/PerfView/PerfViewData.cs

```diff
@@ -3388,6 +3388,436 @@ private void CreateCSVFile(List<AspNetRequest> requests, string csvFileName)
+                outputWriter.WriteLine("</tfoot>");
+                outputWriter.WriteLine("</table>");
+            }
+
+            // slow requests table, only for complete requests
+            // TODO: add command for opening events window
+            if (completeRequests.Count > 0) 
+            {
+                IEnumerable<ANCHostingRequest> slowestRequests =
+                    (from request in completeRequests
+                     where request.DurationMsec >= 100
+                     orderby request.DurationMsec descending
+                     select request).Take(20);
+
+                outputWriter.WriteLine("<h3>Top 20 Slowest Requests (complete requests only)</h3>");
+                outputWriter.WriteLine("<table border=\"1\">");
+                outputWriter.WriteLine("<caption>This table shows the top 20 slowest requests in this trace.<br/>Incomplete requests and requests taking <100 milliseconds are ignored.<br/>Hover over column headings for explanation of columns.</caption>");
+                outputWriter.Write("<thead><tr>");
+                outputWriter.Write("<th align='center' title='Process ID'>PID</th>");
+                outputWriter.Write("<th align='center' title='HTTP Method/Verb - may not be available'>Method</th>");
+                outputWriter.Write("<th align='center' title='Request Path - may not be available'>Path</th>");
+                outputWriter.Write("<th align='center' title='Duration of the request in milliseconds'>Duration (msec)</th>");
+                //outputWriter.Write("<th align='center' title='ActivityId of the request - click to open Events view for this ID'>ActivityID</th>");
+                outputWriter.Write("<th align='center' title='ActivityId of the request - this is a good request-specific text filter for the Events view'>ActivityID</th>");
+                outputWriter.WriteLine("</tr></thead>");
+                outputWriter.WriteLine("<tbody>");
+                if (slowestRequests.Any())
+                {
+                    foreach (ANCHostingRequest request in slowestRequests)
+                    {
+                        // used for opening the request in the Events view
+                        //string detailedRequestCommandString = $"detailedrequestevents:{request.ActivityId};{request.StartTimeRelativeMSec};{request.EndTimeRelativeMSec}";
+
+                        outputWriter.Write("<tr>");
+                        outputWriter.Write($"<td>{request.ProcessID}</td>");
+                        outputWriter.Write($"<td>{request.Method}</td>");
+                        outputWriter.Write($"<td>{request.Path}</td>");
+                        outputWriter.Write($"<td>{request.DurationMsec:N2}</td>");
+                        //outputWriter.Write($"<td><a href=\"command:{detailedRequestCommandString}\">{StartStopActivityComputer.ActivityPathString(request.ActivityId)}</a></td>");
```

Comment:
There are a few extra commented out lines here.  Are these needed?

Reply:
not needed, removed